examples/bri-3/src/bri/transactions/agents/transactionStorage.agent.ts

```diff
@@ -0,0 +1,102 @@
+import { PrismaService } from '../../../../prisma/prisma.service';
+import { NOT_FOUND_ERR_MESSAGE } from '..//api/err.messages';
+import { Transaction } from '../models/transaction';
+
+@Injectable()
+export class TransactionStorageAgent extends PrismaService {
+  async getTransactionById(id: string): Promise<Transaction> {
+    const transactionModel = await this.transaction.findUnique({
+      where: { transactionId: id },
+    });
+
+    if (!transactionModel) {
+      throw new NotFoundException(NOT_FOUND_ERR_MESSAGE);
+    }
+
+    return new Transaction(
+      transactionModel.transactionId,
+      transactionModel.nonce,
+      transactionModel.workflowInstanceId,
+      transactionModel.workstepInstanceId,
+      null, // TODO: transactionModel.from once BpiAccount in the prisma schema,
+      null, // TODO: transactionModel.to once BpiAccount in the prisma schema,
+      transactionModel.payload,
+      transactionModel.signature,
+      transactionModel.status,
+    );
+  }
+
+  async getAllTransactions(): Promise<Transaction[]> {
+    const transactionModels = await this.transaction.findMany();
+    return transactionModels.map((t) => {
+      return new Transaction(
+        t.transactionId,
+        t.nonce,
+        t.workflowInstanceId,
+        t.workstepInstanceId,
+        null, // TODO: transactionModel.from once BpiAccount in the prisma schema,
+        null, // TODO: transactionModel.to once BpiAccount in the prisma schema,
+        t.payload,
+        t.signature,
+        t.status,
+      );
+    });
+  }
+
+  async createNewTransaction(transaction: Transaction): Promise<Transaction> {
+    const newTransactionModel = await this.transaction.create({
+      data: {
+        transactionId: transaction.transactionId,
+        nonce: transaction.nonce,
+        workflowInstanceId: transaction.workflowInstanceId,
+        workstepInstanceId: transaction.workstepInstanceId,
+        from: '', // TODO: transactionModel.from once BpiAccount in the prisma schema,
+        to: '', // TODO: transactionModel.to once BpiAccount in the prisma schema,
+        payload: transaction.payload,
+        signature: transaction.signature,
+        status: transaction.status,
+      },
+    });
+
+    return new Transaction(
+      newTransactionModel.transactionId,
+      newTransactionModel.nonce,
+      newTransactionModel.workflowInstanceId,
+      newTransactionModel.workstepInstanceId,
+      null, // TODO: newTransactionModel.from once BpiAccount in the prisma schema,
+      null, // TODO: newTransactionModel.to once BpiAccount in the prisma schema,
+      newTransactionModel.payload,
+      newTransactionModel.signature,
+      newTransactionModel.status,
+    );
+  }
+
+  async updateTransaction(transaction: Transaction): Promise<Transaction> {
+    const newTransactionModel = await this.transaction.update({
+      where: { transactionId: transaction.transactionId },
+      data: {
+        payload: transaction.payload,
```

Comment:
why are only these 2 updatable? maybe then we should pass only those 2 here or?

Reply:
It is still unknown what will be updatable and in which part of the transaction lifecycle, so i just added these as an example. We ll expand on this once we move into the transaction execution implementation in later phases of the project